Test thread_flood is failing since we have a new thread "usbus"

#### Description
Test thread_flood is a simple test to see if the maximum number of tests can be created. It counts the number of threads created and subtracts two (for "main" and "idle") and checks if that counts up to `MAXTHREADS`. However we now can also have a thread for "usbus", so the test is not correct anymore.

#### Steps to reproduce the issue
```
BOARD=sodaq-autonomo make -C tests/thread_flood flash term
```
This can be any board with USBUS.

#### Expected results
```
[SUCCESS] created 30
```

#### Actual results
```
[ERROR] expected 30, created 29

--- a/tests/thread_flood/main.c
+++ b/tests/thread_flood/main.c
@@ -39,6 +39,7 @@ int main(void)
 {
     kernel_pid_t thr_id = KERNEL_PID_UNDEF;
     unsigned thr_cnt = 0;
+    unsigned thr_in_use = sched_num_threads;
 
     puts("[START] Spawning threads");
     do {
@@ -54,12 +55,11 @@ int main(void)
     /* decrease by 1 because last thread_create failed */
     --thr_cnt;
 
-    /* expect (MAXTHREADS - 2), as main and idle thread take a PID each */
-    if (thr_cnt == (MAXTHREADS - 2)) {
+    if (thr_cnt == (MAXTHREADS - thr_in_use)) {
         printf("[SUCCESS]");
     }
     else {
-        printf("[ERROR] expected %u,", (MAXTHREADS - 2));
+        printf("[ERROR] expected %u,", (MAXTHREADS - thr_in_use));
     }
     printf(" created %u\n", thr_cnt);
 

